core/src/main/scala/chisel3/internal/firrtl/IR.scala

```diff
@@ -171,7 +171,9 @@ private[chisel3] object ir {
       val unsigned = if (n < 0) (BigInt(1) << width.get) + n else n
       s"asSInt(${ULit(unsigned, width).name})"
     }
-    def minWidth: Int = (if (w.known) 0 else 1) + n.bitLength
+
+    // Special case for 0 which can be specified to zero-width (but defaults to 1 bit).
+    def minWidth: Int = if (n == 0) (if (w.known) 0 else 1) else 1 + n.bitLength
```

Comment:
I would only consider the following change if you think it better calls out the truly special case. If `!w.known`, then minWidth behaves like anything else: 

```suggestion
    // Special case for 0 which can be specified to zero-width (but defaults to 1 bit).
    def minWidth: Int = if (n == 0 && w.known) 0 else 1 + n.bitLength
```

Only for consideration.

Reply:
I do actually think that's better, thanks! I didn't love the nested if.